eldritch/stories.py

```diff
@@ -0,0 +1,202 @@
+  @abc.abstractmethod
+  def __init__(self, name, pass_result, fail_result):
+    super().__init__(name, None, "specials", active_bonuses={}, passive_bonuses={})
+    self.results = {True: pass_result, False: fail_result}
+
+  def advance_story(self, character, pass_story):
+    return events.Sequence([
+        events.DiscardSpecific(character, [self]),
+        events.DrawSpecific(character, "specials", self.results[pass_story])
+    ], character)
+
+
+class DrifterStoryPass(StoryResult):
+  def __init__(self):
+    super().__init__("Sweet Dreams", passive_bonuses={"speed": 1, "luck": 1})
+
+
+class DrifterStoryFail(StoryResult):
+  def __init__(self):
+    super().__init__("Living Nightmare")
+
+  def get_in_play_event(self, owner: characters.Character):
+    allies = [
+        ally
+        for ally in owner.possessions
+        if getattr(ally, "deck", None) == "allies" and ally.name != "Dog"
+    ]
+    return events.DiscardSpecific(owner, allies)
+
+  def get_interrupt(self, event, owner, state):
+    if not getattr(event, "character", None) == owner:
+      return super().get_interrupt(event, owner, state)
+    if (
+        (isinstance(event, events.KeepDrawn)
+         and (getattr(event.drawn, "deck", None) == "allies"))
+        or (isinstance(event, (events.DrawNamed, events.DrawItems)) and event.deck == "allies")
+    ):
+      return events.CancelEvent(event)
+    return super().get_interrupt(event, owner, state)
+
+
+class DrifterStory(Story):
+  def __init__(self):
+    super().__init__(
+        "Powerful Nightmares", "Sweet Dreams", "Living Nightmare"
+    )
+
+  def get_trigger(self, event, owner, state):
+    if (
+        isinstance(event, events.TakeGateTrophy)
+        and (event.character == owner)
+        and (event.gate.name == "Dreamlands")
+    ):
+      return self.advance_story(owner, True)
+    if (
+        isinstance(event, events.AddDoom)
+        and state.ancient_one.doom >= 5
+    ):
+      return self.advance_story(owner, False)
+    return super().get_trigger(event, owner, state)
+
+
+class GangsterStoryPass(StoryResult):
+  def __init__(self):
+    super().__init__("This One's for Louis", {"max_sanity": 1})
+
+  def get_in_play_event(self, owner: characters.Character):
+    print("Gangster Story Pass now in play")
+    return events.Gain(owner, {"sanity": 1}, self)
+
+  def get_trigger(self, event, owner, state):
+    if isinstance(event, events.TakeTrophy) and event.character == owner:
+      return events.Gain(owner, {"sanity": 1}, self)
+    return super().get_trigger(event, owner, state)
+
+
+class GangsterStoryFail(StoryResult):
+  def __init__(self):
+    super().__init__("Crime Doesn't Pay")
+
+  def get_in_play_event(self, owner: characters.Character):
+    return events.Loss(owner, {"dollars": owner.dollars})
+
+  def get_interrupt(self, event, owner, state):
+    if (
+        isinstance(event, events.GainOrLoss)
+        and event.character == owner
+        and "dollars" in event.gains
```

Comment:
Do we think this also affects trading money?

Reply:
Ooh, it probably does. I see there's a TODO on the trading code to make it an event, so we can either do that as part of this PR, or leave it in the to do list.